nano/node/node.cpp

```diff
@@ -103,6 +103,22 @@ void nano::network::send_keepalive (nano::transport::channel const & channel_a)
 	channel_a.send (message);
 }
 
+void nano::network::send_keepalive_self (nano::transport::channel const & channel_a)
+{
+	nano::keepalive message;
+	udp_channels.random_fill (message.peers);
+	if (node.config.external_address != boost::asio::ip::address_v6{} && node.config.external_port != 0)
+	{
+		message.peers[0] = nano::endpoint (node.config.external_address, node.config.external_port);
+	}
+	else
+	{
+		message.peers[0] = nano::endpoint (boost::asio::ip::address_v6{}, endpoint ().port ());
+		message.peers[1] = nano::endpoint (boost::asio::ip::address_v6{}, node.port_mapping.external_address ().port ());
```

Comment:
UPnP also reports external ip. Do we wanna use that?

Reply:
We can use it as well, yes
To prevent possible upnp errors `message.peers[2] = node.port_mapping.external_address ();`